timelock-impl/src/main/java/com/palantir/atlasdb/timelock/transaction/client/DistributingModulusGenerator.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.transaction.client;
+
+import java.util.Comparator;
+import java.util.SortedSet;
+import java.util.stream.IntStream;
+
+import org.immutables.value.Value;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+
+/**
+ * Distributes residues of a given modulus in a balanced fashion (though we don't automatically rebalance between
+ * residues).
+ *
+ * Unmarking residues may not be performant if the number of residues used is large.
+ */
+public class DistributingModulusGenerator {
+    private final SortedSet<ReferenceCountedResidue> referenceCounts;
+
+    @VisibleForTesting
+    DistributingModulusGenerator(SortedSet<ReferenceCountedResidue> referenceCounts) {
+        this.referenceCounts = referenceCounts;
```

Comment:
whilst this is `@VisibleForTesting` you're not actually using it, you're using `create` below. I think the constructor should take the `modulus` as opposed to an arbitrary `SortedSet` i.e. you could write a test with a sorted set sorted arbitrarily as opposed to how you actually want it to be sorted => impl is in the test.